modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/search/filter/FilterUtil.java

```diff
@@ -14,40 +14,245 @@
+import com.liferay.portal.kernel.search.filter.BooleanFilter;
+import com.liferay.portal.kernel.search.filter.DateRangeTermFilter;
+import com.liferay.portal.kernel.search.filter.ExistsFilter;
 import com.liferay.portal.kernel.search.filter.Filter;
+import com.liferay.portal.kernel.search.filter.FilterVisitor;
+import com.liferay.portal.kernel.search.filter.GeoBoundingBoxFilter;
+import com.liferay.portal.kernel.search.filter.GeoDistanceFilter;
+import com.liferay.portal.kernel.search.filter.GeoDistanceRangeFilter;
+import com.liferay.portal.kernel.search.filter.GeoPolygonFilter;
+import com.liferay.portal.kernel.search.filter.MissingFilter;
+import com.liferay.portal.kernel.search.filter.PrefixFilter;
+import com.liferay.portal.kernel.search.filter.QueryFilter;
+import com.liferay.portal.kernel.search.filter.RangeTermFilter;
 import com.liferay.portal.kernel.search.filter.TermFilter;
-import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.search.filter.TermsFilter;
+import com.liferay.portal.kernel.search.generic.BooleanQueryImpl;
+import com.liferay.portal.kernel.search.generic.NestedQuery;
+import com.liferay.portal.kernel.search.generic.TermQueryImpl;
+import com.liferay.portal.kernel.search.generic.TermRangeQueryImpl;
+import com.liferay.portal.kernel.search.generic.WildcardQueryImpl;
+
+import java.util.List;
 
 /**
  * @author Javier de Arcos
  */
 public class FilterUtil {
 
-	public static Filter processFilter(DDMIndexer ddmIndexer, Filter filter)
-		throws Exception {
+	public static Filter processFilter(DDMIndexer ddmIndexer, Filter filter) {
+		if ((filter == null) || ddmIndexer.isLegacyDDMIndexFieldsEnabled()) {
+			return filter;
+		}
+
+		DDMFilterVisitor ddmFilterVisitor = new DDMFilterVisitor(ddmIndexer);
+
+		return filter.accept(ddmFilterVisitor);
+	}
 
-		if (ddmIndexer.isLegacyDDMIndexFieldsEnabled() ||
-			!(filter instanceof TermFilter)) {
+	private static class DDMFilterVisitor implements FilterVisitor<Filter> {
 
-			return filter;
+		public DDMFilterVisitor(DDMIndexer ddmIndexer) {
+			_ddmIndexer = ddmIndexer;
 		}
 
-		TermFilter termFilter = (TermFilter)filter;
+		@Override
+		public Filter visit(BooleanFilter booleanFilter) {
+			BooleanFilter transformedBooleanFilter = new BooleanFilter();
 
-		String termFilterField = termFilter.getField();
+			_addTransformedFilters(
+				booleanFilter.getMustBooleanClauses(), BooleanClauseOccur.MUST,
+				transformedBooleanFilter);
+			_addTransformedFilters(
+				booleanFilter.getMustNotBooleanClauses(),
+				BooleanClauseOccur.MUST_NOT, transformedBooleanFilter);
+			_addTransformedFilters(
+				booleanFilter.getShouldBooleanClauses(),
+				BooleanClauseOccur.SHOULD, transformedBooleanFilter);
 
-		if (!termFilterField.startsWith(DDMIndexer.DDM_FIELD_PREFIX)) {
-			return filter;
+			return transformedBooleanFilter;
+		}
+
+		@Override
+		public Filter visit(DateRangeTermFilter dateRangeTermFilter) {
+			return dateRangeTermFilter;
```

Comment:
is this working right now? because it should from a product side of things

Reply:
I've made tests with dates and it is working because the filter is modeled as a RangeTermFilter and not as a DateRangeTermFilter. I've tried to implement it but the DateRangeFilterImpl clause is internal and I can't create a new instance with the new field value

Reply:
Sigh, Ok. We should document it somewhere if it has effects on how to filter. I don't think it's very important to justify a technical debt if noone complains (that I think they won't ever) but if we do we can copy the behaviour/use reflection.